scarb/src/ops/resolve.rs

```diff
@@ -619,9 +626,13 @@ impl<'a> PackageSolutionCollector<'a> {
                 // We can safely unwrap as all packages with `PackageClass::CairoPlugin` must define plugin target.
                 let target = package.target(&TargetKind::CAIRO_PLUGIN).unwrap();
                 let props: CairoPluginProps = target.props()?;
+                let prebuilt = ProcMacroInstance::try_load_prebuilt(package.clone())
```

Comment:
I am not sure, if loading the same dynamic library into the memory multiple times is legal (especially on windows). For safety, I would do it in two steps 
- First generate all CairoCompilationUnits
- Then iter through all cairo_plugins and create ProceduralMacroCompilationUnits for them (loading the plugins)
- And then, reiterate through CairoCompilationUnits, cloning the Arc with loaded macro into each maching CairoCompilationUnitComponent. 

This way we do not need to load any macro twice. 
Also, it might be worth the effort, to replicate this situation in tests - i.e. instead of a single package, have two packages relying on the same plugin in a single workspace.